### PR TITLE
Yautja Falcon Drone

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Yautja now have a falcon drone, a surveillance gadget that lets them scout the hunting grounds. Makes use of the queen eye code as a basis for its mechanics.

## Why It's Good For The Game

It has radiant soul. It's a gadget found in some of the official video games.

## Changelog

:cl: Geeves, monkeysfist101
add: Yautja now have a falcon drone, a surveillance gadget that lets them scout the hunting grounds.
expansion: You can now examine when your vision is detached from your body (queen eye, falcon drone, etc.)
/:cl:

### CODE REVIEW

**Comment by johnravioliUSA** [comment]
Soulful ASF Speedmerge NOW!!!

**Comment by Waseemq1235** [comment]
Drone seems completely untrackable via the Track Yautja Gear verb, regardless if it is undeployed, deployed, or destroyed.

**Comment by Geevies** [comment]
> Drone seems completely untrackable via the Track Yautja Gear verb, regardless if it is undeployed, deployed, or destroyed.

I wanna overhaul the tracking thing for Yautja by tracking anything with the ITEM_PREDATOR flag automatically, because at the moment I need to add it to each item manually which is a pain. So I'll say it's out of scope for now.

**Comment by Waseemq1235** [comment]
> > Drone seems completely untrackable via the Track Yautja Gear verb, regardless if it is undeployed, deployed, or destroyed.
> 
> I wanna overhaul the tracking thing for Yautja by tracking anything with the ITEM_PREDATOR flag automatically, because at the moment I need to add it to each item manually which is a pain. So I'll say it's out of scope for now.

I don't know if marines getting an untrackable drone for themselves is a good idea, unless they simply can't use it, in which case, yeah nice.

**Review by Agameofscones — APPROVED** [review]
Sprites LGTM

File: code/modules/cm_preds/falcon.dm [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			return "on [t_his] shoulder"
+		if(WEAR_R_EAR)
+			return "on [t_his] shoulder"
+	return ..()
+
+/obj/item/falcon_drone/verb/control_falcon_drone()
+	set name = "Control Falcon Drone"
+	set desc = "Activates your falcon drone."
+	set category = "Yautja.Misc"
+	set src in usr
```
Comment (stanalbatross) [inline code/modules/cm_preds/falcon.dm:27]: make this also trigger on use_hand

also slightly confusing UX that you use a verb to start using the drone and an action button to stop using it. Ideally both would use action buttons.

File: code/modules/cm_preds/falcon.dm [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+/mob/hologram/falcon/ex_act()
+	new /obj/item/trash/falcon_drone(loc)
+	QDEL_NULL(parent_drone)
+	qdel(src)
+
+/mob/hologram/falcon/emp_act()
+	new /obj/item/trash/falcon_drone/emp(loc)
+	QDEL_NULL(parent_drone)
+	qdel(src)
```
Comment (stanalbatross) [inline code/modules/cm_preds/falcon.dm:91]: can humans even see it? I couldn't during local testing

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/cm_preds/falcon.dm (hunk 2 above)
Comment (Geevies) [reply code/modules/cm_preds/falcon.dm:91]: nope, you can only see em with a pred mask on

File: code/modules/cm_preds/falcon.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/cm_preds/falcon.dm:30]: added the in-hand, and eh. the action buttons aren't really needed for a once-in-a-while drone

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**